src/vs/workbench/api/browser/mainThreadTerminalShellIntegration.ts

```diff
@@ -43,6 +43,20 @@ export class MainThreadTerminalShellIntegration extends Disposable implements Ma
 		})).event;
 		this._store.add(onDidAddCommandDetection(e => this._proxy.$shellIntegrationChange(e.instanceId)));
 
+		// onDidChangeTerminalShellIntegration via cwd
+		const cwdChangeEvent = this._store.add(this._terminalService.createOnInstanceCapabilityEvent(TerminalCapability.CwdDetection, e => e.onDidChangeCwd));
+		this._store.add(cwdChangeEvent.event(e => {
+			this._proxy.$cwdChange(e.instance.instanceId, this._convertCwdToUri(e.data));
+		}));
+
+		// onDidChangeTerminalShellIntegrationEnvironment
```

Comment:
This comment calls out actual APIs and we don't have an environment one atm.

```suggestion
		// onDidChangeTerminalShellIntegration via env
```